src/vcCamera.cpp

```diff
@@ -252,11 +252,26 @@ void vcCamera_Apply(vcState *pProgramState, vcViewport *pViewport, vcCameraSetti
     double vertPos = addPos.z;
 
     addPos.z = 0.0;
-    addPos = orientation.apply(addPos);
 
     // Translation
-    if ((pCamSettings->mapMode[pProgramState->activeViewportIndex] || pCamSettings->lockAltitude) && addPos != udDouble3::zero())
-      addPos -= udDot(pViewport->camera.cameraUp, addPos) * pViewport->camera.cameraUp;
+    if (pCamSettings->mapMode[pProgramState->activeViewportIndex])
+    {
+      if (pProgramState->geozone.projection == udGZPT_ECEF)
```

Comment:
This PR is wrong, I'm not 100% sure how but if you need to check the projection- it's wrong.